File: src/main/java/Jikan4java/types/Support/Related/Related.java
package Jikan4java.types.Support.Related;
/**
 * Jikan4java
 * 28 / October / 2018
 *
 * @author github.com/doomsdayrs
 */

/*
This file is part of Jikan4java.

Jikan4java is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Foobar is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
*/

import Jikan4java.types.Support.Related.Types.*;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;

public class Related {

    @JsonProperty("prequel")
    private ArrayList<Prequel> prequel;

    @JsonProperty("Alternative version")
    private ArrayList<AltVersion> altVersion;

    @JsonProperty("Spin-off")
    private ArrayList<SpinOff> spinOff;

    @JsonProperty("Adaptation")
    private ArrayList<Adaptation> adaptation;

    @JsonProperty("Summary")
    private ArrayList<Summary> summaries;

    @JsonProperty("Sequel")
    private ArrayList<Sequel> sequels;

    @JsonProperty("Side story")
    private ArrayList<SideStory> sideStories;

    @JsonProperty("Other")
    private ArrayList<Other> other;

    public ArrayList<Prequel> getPrequel() {
        return prequel;
    }

    public ArrayList<AltVersion> getAltVersion() {
        return altVersion;
    }

    public ArrayList<SpinOff> getSpinOff() {
        return spinOff;
    }

    public ArrayList<Adaptation> getAdaptation() {
        return adaptation;
    }

    public ArrayList<Summary> getSummaries() {
        return summaries;
    }

    public ArrayList<Sequel> getSequels() {
        return sequels;
    }

    public ArrayList<SideStory> getSideStories() {
        return sideStories;
    }

    public ArrayList<Other> getOther() {
        return other;
    }
}

File: src/commonMain/kotlin/com/github/doomsdayrs/jikan4java/data/model/support/related/Related.kt
package com.github.doomsdayrs.jikan4java.data.model.support.related

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================
 * Jikan4java
 * 28 / October / 2018
 *
 * @author github.com/doomsdayrs
 */
@Serializable
data class Related(
	@SerialName("Prequel")
	val prequel: List<RelatedType> = listOf(),

	@SerialName("Alternative version")
	val altVersion: List<RelatedType> = listOf(),

	@SerialName("Spin-off")
	val spinOff: List<RelatedType> = listOf(),

	@SerialName("Adaptation")
	val adaptation: List<RelatedType> = listOf(),

	@SerialName("Summary")
	val summaries: List<RelatedType> = listOf(),

	@SerialName("Sequel")
	val sequels: List<RelatedType> = listOf(),

	@SerialName("Side story")
	val sideStories: List<RelatedType> = listOf(),

	@SerialName("Other")
	val other: List<RelatedType> = listOf(),

	@SerialName("Parent story")
	val parentStories: List<RelatedType> = listOf(),

	@SerialName("Alternative setting")
	val alternativeSettings: List<RelatedType> = listOf(),

	@SerialName("Character")
	val characters: List<RelatedType> = listOf(),

	@SerialName("Full story")
	val fullStories: List<RelatedType> = listOf()
)